service/src/test/java/com/kodemon/service/test/facade/UserFacadeTest.java

```diff
@@ -0,0 +1,137 @@
+
+        trainer2 = new Trainer();
+        trainer2.setUserName("blin5");
+        trainer2.setFirstName("Adam");
+        trainer2.setLastName("Blin");
+        born = new Calendar.Builder().setDate(1995, 4, 8).build().getTime();
+        trainer2.setDateOfBirth(born);
+
+        user = new UserDTO();
+        user.setUserName("brock1999");
+        user.setFirstName("Brock");
+        user.setLastName("Brockowski");
+        born = new Calendar.Builder().setDate(1999, 5, 5).build().getTime();
+        user.setDateOfBirth(born);
+
+        user2 = new UserDTO();
+        user2.setUserName("blin5");
+        user2.setFirstName("Adam");
+        user2.setLastName("Blin");
+        born = new Calendar.Builder().setDate(1995, 4, 8).build().getTime();
+        user2.setDateOfBirth(born);
+    }
+
+    @Test
+    public void registerTest() {
+        when(beanMappingService.mapTo(user, Trainer.class)).thenReturn(trainer);
+        when(trainerService.register(trainer, "password")).thenReturn(true);
+        boolean result = userFacade.register(user, "password");
+        assertThat(result, is(true));
+    }
+
+    @Test
+    public void loginTest() {
+        when(beanMappingService.mapTo(user, Trainer.class)).thenReturn(trainer);
+        when(trainerService.register(trainer, "password")).thenReturn(true);
+        boolean prerequisite = userFacade.register(user, "password");
+        assertThat(prerequisite, is(true));
+
+        UserAuthDTO userAuth = new UserAuthDTO();
+        userAuth.setUserName(trainer.getUserName());
+        userAuth.setPwdHash(trainer.getPwdHash());
+
+        when(trainerService.login(userAuth.getUserName(), userAuth.getPwdHash())).thenReturn(true);
+        boolean result = userFacade.login(userAuth);
+        assertThat(result, is(true));
+    }
+
+    @Test
+    public void findUserByUserNameTest() {
+        List<Trainer> trainerListE = new ArrayList<>();
+        trainerListE.add(trainer);
+        List<UserDTO> trainerList = new ArrayList<>();
+        trainerList.add(user);
+        when(trainerService.findByUserName(trainer.getUserName())).thenReturn(trainerListE);
+        when(beanMappingService.mapTo(trainerListE, UserDTO.class)).thenReturn(trainerList);
+        List<UserDTO> result = userFacade.findUserByUserName(user.getUserName());
+        assertThat(result.size(), is(1));
+        assertThat(result.get(0), is(user));
+    }
+
+    @Test
+    public void findAllUsersTest() {
+        List<Trainer> allTrainersE = new ArrayList<>();
+        allTrainersE.add(trainer);
+        allTrainersE.add(trainer2);
+        List<UserDTO> allTrainers = new ArrayList<>();
+        allTrainers.add(user);
+        allTrainers.add(user2);
+        when(trainerService.findAll()).thenReturn(allTrainersE);
+        when(beanMappingService.mapTo(allTrainersE, UserDTO.class)).thenReturn(allTrainers);
+        List<UserDTO> result = userFacade.findAllUsers();
+        assertThat(result.size(), is(2));
```

Comment:
Maybe you should also assert that the list is correct.. eg. the DTOs in the list are the wanted ones.